File: hackhive-frontend/src/services/projectResultService.js
import api from "../api/axios";

async function request(promise) {
    const response = await promise;
    return response.data?.data ?? response.data;
}

export const projectResultService = {
    configureResult: (eventId, data) =>
        request(api.post(`/project-results/events/${eventId}`, data)),

    getEventLeaderboardForOrganizer: (eventId) =>
        request(api.get(`/project-results/events/${eventId}`)),

    updateResult: (resultId, data) =>
        request(api.put(`/project-results/${resultId}`, data)),

    publishResult: (resultId) =>
        request(api.post(`/project-results/${resultId}/publish`)),

    publishAllResultsForEvent: (eventId) =>
        request(api.post(`/project-results/events/${eventId}/publish-all`)),

    getPublishedResultsForEvent: (eventId) =>
        request(api.get(`/project-results/events/${eventId}/published`)),

    getPublishedResultForTeam: (teamId) =>
        request(api.get(`/project-results/team/${teamId}/published`)),
};

export default projectResultService;

File: hackhive-backend/src/main/java/com/hackhive/submission/service/ProjectResultService.java
package com.hackhive.submission.service;

import com.hackhive.submission.dto.request.ConfigureProjectResultRequest;
import com.hackhive.submission.dto.request.UpdateProjectResultRequest;
import com.hackhive.submission.dto.response.ProjectResultResponse;

import java.util.List;

public interface ProjectResultService {

    ProjectResultResponse configureResult(Long eventId, ConfigureProjectResultRequest request);

    List<ProjectResultResponse> getEventLeaderboardForOrganizer(Long eventId);

    ProjectResultResponse updateResult(Long resultId, UpdateProjectResultRequest request);

    ProjectResultResponse publishResult(Long resultId);

    List<ProjectResultResponse> publishAllResultsForEvent(Long eventId);

    List<ProjectResultResponse> getPublishedResultsForEvent(Long eventId);

    ProjectResultResponse getPublishedResultForTeam(Long teamId);
}
